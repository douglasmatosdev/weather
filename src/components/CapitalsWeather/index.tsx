import React from 'react'
import styled from 'styled-components'
import { ListForecast } from '../../types'

interface Props {
    capitals: ListForecast[]
}

export default function CapitalsWeather({ capitals }: Props): JSX.Element {

    return (
        <CapitalsWeatherContainer>
            <h2>Capitais</h2>
            <div className="capitais">
                <ul>
                    <h5>Min Max</h5>
                    {capitals?.map((e, i) => {
                        return (
                            <li key={i}>
                                <span className="min-max">
                                    {e?.main?.temp_min?.toFixed(0)}°
                                    {e?.main?.temp_max?.toFixed(0)}°
                                </span>
                                <span className="cidade">{e?.name}</span>
                            </li>
                        )
                    })}
                </ul>
            </div>

        </CapitalsWeatherContainer>
    )
}

const CapitalsWeatherContainer = styled.section`
    padding: 24px;
    h2 {
        font-size: 32px;
        font-weight: 600;
        line-height: 32px;
        color: #fff;
    }

    .capitais {
        h5 {
            font-size: 13px;
            font-weight: 100;
            margin: 24px 0px 16px 0px;
        }
        &, ul {
            display: flex;
        }
        flex-wrap: wrap;
        ul {
            width: 100%;
            /* border: 1px solid #000; */
            flex-direction: column;

            li {
                padding: 8px 0px;

                .min-max {
                    margin-right: 16px;
                }
            }
            li span {
                font-size: 16px;
                font-weight: 600;
                
                
            }
        }
    }
`