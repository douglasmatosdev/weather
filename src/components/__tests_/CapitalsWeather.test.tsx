import React from "react";
import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import CapitalsWeather from "../CapitalsWeather";

import capitals from "../../utils/idCapitalsBrazil";

// Should be to match snapshot
test("Should be rendered without crash", () => {
    const component = render(<CapitalsWeather capitals={[]} />);

    expect(component.container).toMatchSnapshot();
});

// Should be have two colums
test("Should be have two colums", () => {
    const { getByTestId } = render(<CapitalsWeather capitals={[]} />);
});

// Should be have ten list item
test("Should be have ten list item", () => { });

// Cannot have repeat items
test("Cannot have repeat items", () => { });