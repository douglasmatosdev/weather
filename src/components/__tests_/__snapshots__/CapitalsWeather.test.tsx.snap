// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should be rendered without crash 1`] = `
<div>
  <section
    class="sc-bdnxRM bhqYwV"
  >
    <h2>
      Capitais
    </h2>
    <div
      class="capitals"
    >
      <ul
        id="title-columns"
      >
        <li>
          Min Max
        </li>
        <li>
          Min Max
        </li>
      </ul>
      <ul
        id="capitals-columns"
      />
    </div>
  </section>
</div>
`;
